web-service/src/main/java/com/winsel/service/TaskService.java

```diff
@@ -5,31 +5,44 @@
 import com.winsel.dao.entity.TaskType;
 import com.winsel.dao.entity.User;
 import com.winsel.dao.entity.WeatherTask;
+import com.winsel.dto.WeatherResponse;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
 
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import java.util.ArrayList;
-import java.util.List;
 import java.util.Optional;
 
 @Service
 public class TaskService {
+    @Value("${app.apiLink}")
+    private String apiLink;
     @Autowired
     private TaskRepository taskRepository;
     @Autowired
     private TaskTypeRepository taskTypeRepository;
+    @Autowired
+    private WeatherTaskRepository weatherTaskRepository;
+
+    public WeatherResponse weatherApi(LocalDateTime localDateTime){
+        RestTemplate restTemplate = new RestTemplate();
+        return restTemplate.getForObject(apiLink+localDateTime, WeatherResponse.class);
```

Comment:
Change this for

#Assume apiLink = "http://host:port"
URI targetUrl= UriComponentsBuilder.fromUriString(apiLink)
    .path("/weather")
    .queryParam("localDateTime", localDateTime)
    .build()
    .encode()
    .toUri();

return restTemplate.getForObject(targetUrl, WeatherResponse.class);